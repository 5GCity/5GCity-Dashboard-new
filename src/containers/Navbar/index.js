/**
 * Navbar Container
 * Please write a description
 *
 * @author Your Name <gpatriarca@ubiwhere.com>
 */
import React, { Component } from 'react'
import Logic from './logic'
import styled from 'styled-components'
import { withRouter } from 'react-router-dom'
import { Icon } from 'element-react'

/* Components */
import Brand from 'components/Brand'
import Modal from 'components/Modal'
import Button from 'components/Button'
import SideBar from 'components/SideBar'
import { CheckIcon, CloseIcon } from 'components/Icons'

class Navbar extends Component {
  render () {
    const { modalChangeStatus, logout, changeLink } = this.actions
    const { userName, userRole, modalStatus, links } = this.props
    return (
      <Wrapper>
        <Modal
          visible={modalStatus}
          title={'Are you sure you want to leave?'}
          size={'tiny'}
          showClose={false}
          onCancel={modalChangeStatus}
        >
          <Modal.Body />
          <Modal.Footer>
            <Button
              text={'Yes'}
              type={'primary'}
              svg={<CheckIcon />}
              onClick={logout}
            />
            <Button
              text={'No'}
              type={'danger'}
              svg={<CloseIcon />}
              onClick={modalChangeStatus}
            />
          </Modal.Footer>
        </Modal>
        <SideBar>
          <Brand />
          {links && links.map(menu =>
            <SideBar.Container key={menu.id}>
              <SideBar.Menu
                height={96}
                onClick={() => changeLink(menu)}
                active={menu.active}
                disabled={menu.disabled}
              >
              {menu.icon &&
                <SideBar.Icon>{menu.icon}</SideBar.Icon>
              }
                <SideBar.Link>{menu.name}</SideBar.Link>
              {menu.children && !menu.active &&
                <Caret name={'caret-bottom'} />
              }
              {menu.children && menu.active &&
                <Caret name={'caret-top'} />
              }
              </SideBar.Menu>
              {menu.children && menu.children.map((submenu, i) =>
                <SideBar.SubMenu
                  key={i}
                  show={menu.active}
                  onClick={() => changeLink(submenu)}
                  disabled={submenu.disabled}
                  active={submenu.active}
                >
                  <SideBar.Link>{submenu.name}</SideBar.Link>
                </SideBar.SubMenu>
              )}
            </SideBar.Container>
          )}
          <SideBar.UserInfo>
            <SideBar.UserName>{userName}</SideBar.UserName>
            <SideBar.UserRole>{userRole}</SideBar.UserRole>
            <SideBar.IconUser
              className='el-icon-more'
              onClick={modalChangeStatus}
             />
          </SideBar.UserInfo>
        </SideBar>
      </Wrapper>
    )
  }
}

export default withRouter(Logic(Navbar))

const Wrapper = styled.div`
position: fixed;
left: 0;
top: 0;
height: 100vh;
width: 120px;
z-index: 1000;
`
const Caret = styled(Icon)`
  padding-top: 2px;
  color:white;
  font-size: 12px;
`
