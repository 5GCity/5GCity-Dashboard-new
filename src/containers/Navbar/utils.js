/**
 * Navbar Container Utils
 * Please write a description
 *
 * @author Your Name <youremail@ubiwhere.com>
 */

export const giveUserRole = (userArray) => {
    const findAdmin = userArray.roles.find(role => role === 'Administrator')
    return findAdmin !== undefined ? 'Inf. Owner' : 'Slice Requester'
 }