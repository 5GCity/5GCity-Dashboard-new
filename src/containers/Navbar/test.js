/**
 * Navbar Container Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Navbar from './index'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import { getStore } from 'kea'
import { MemoryRouter } from 'react-router-dom'

const store = getStore()

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Navbar />
       </MemoryRouter>
      </Provider>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
