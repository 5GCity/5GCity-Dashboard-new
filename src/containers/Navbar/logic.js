/**
 * Navbar Container Logic
 * Please write a description
 *
 * @author Your Name <gpatriarca@ubiwhere.com>
 */

import { kea } from 'kea'
import { put } from 'redux-saga/effects'
import PropTypes from 'prop-types'
import AppLogic from 'containers/App/logic'
import { giveUserRole } from './utils'

export default kea({
  path: () => ['scenes', 'containers', 'Navbar'],

  connect: {
    props: [
      AppLogic, [
        'keycloak'
      ]
    ]
  },

  actions: () => ({
    setMsgLogout: () => ({ }),
    closeModal:() => ({ }),
    logout: () => ({ }),
  }),

  reducers: ({ actions }) => ({
    showModal: [false, PropTypes.boolean,{
      [actions.setMsgLogout]: (state, payload) => true,
      [actions.closeModal]: (state, payload) => false
    }]
  }),
  
  selectors: ({ selectors }) => ({
    userName: [
      () => [selectors.keycloak],
      (keycloak) =>{ return keycloak.tokenParsed.name },
      PropTypes.string
    ],
    userRole: [
      () => [selectors.keycloak],
      (keycloak) =>{ return giveUserRole(keycloak.tokenParsed.realm_access) },
      PropTypes.any
    ],
  }),

  takeLatest: ({ actions, workers }) => ({
    [actions.logout]:workers.logout
  }),

  start: function * () {

  },

  workers: {
    * logout(){
      const { closeModal } = this.actions
      yield put(closeModal())
      const keycloak = yield this.get('keycloak')
      console.log(keycloak)
      keycloak.logout()        
    }
  }
})

