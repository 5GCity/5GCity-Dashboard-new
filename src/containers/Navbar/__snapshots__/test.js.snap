// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: fixed;
  left: 0;
  top: 0;
  height: 100vh;
  width: 120px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: -5px 0 20px 0 rgba(0,0,0,0.5);
  z-index: 1000;
}

.c1 {
  width: 100%;
  height: 88px;
  background-color: #FFFFFF;
  box-shadow: inset 0 -1px 0 0 rgba(137,151,159,0.15);
}

.c2 {
  text-decoration: none;
  color: white;
  text-align: center;
  cursor: pointer;
  height: 92px;
  padding: 14px 16px;
  box-shadow: inset 0 -1px 0 0 rgba(137,151,159,0.15);
}

.c3 {
  font-size: 32px;
}

.c4 {
  line-height: 15px;
  padding-top: 8px;
  font-size: 14px;
  text-transform: uppercase;
}

<div
  className="c0"
>
  <div
    className="c1"
    icon=""
  />
  <a
    className="c2"
    disabled={false}
    href="/"
    onClick={[Function]}
  >
    <i
      className="el-icon-edit c3"
      style={Object {}}
    />
    <div
      className="c4"
    >
      Monitoring Dashboard
    </div>
  </a>
  <a
    className="c2"
    disabled={false}
    href="/slices"
    onClick={[Function]}
  >
    <i
      className="el-icon-edit c3"
      style={Object {}}
    />
    <div
      className="c4"
    >
      Slices
    </div>
  </a>
  <a
    className="c2"
    disabled={false}
    href="/network"
    onClick={[Function]}
  >
    <i
      className="el-icon-edit c3"
      style={Object {}}
    />
    <div
      className="c4"
    >
      Network Services
    </div>
  </a>
  <a
    className="c2"
    disabled={false}
    href="/settings"
    onClick={[Function]}
  >
    <i
      className="el-icon-edit c3"
      style={Object {}}
    />
    <div
      className="c4"
    >
      Settings
    </div>
  </a>
</div>
`;
